### PR TITLE
fix: Test for ipset comment support

### LINKED ISSUE #3098: weaveworks/weave-npc:2.0.3 image crashing on startup
## What you expected to happen?
No crash

## What happened?
weave-npc cause CrashLoopBackoff in weave-net pod

## How to reproduce it?
kubectl apply -f https://cloud.weave.works/k8s/net?k8s-version=v1.7.4

## Anything else we need to know?
This is running on a single node kubernetes cluster created with kubeadm ala https://kubernetes.io/docs/setup/independent/create-cluster-kubeadm/

## Versions:
```
$ docker version
Client:
 Version:      17.06.1-ce
 API version:  1.30
 Go version:   go1.8.3
 Git commit:   874a737
 Built:        Thu Aug 17 22:53:49 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.06.1-ce
 API version:  1.30 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   874a737
 Built:        Thu Aug 17 23:01:50 2017
 OS/Arch:      linux/amd64
 Experimental: false
$ uname -a
Linux wshu-docker 3.10.0-514.26.2.el7.x86_64 #1 SMP Tue Jul 4 15:04:05 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.4", GitCommit:"793658f2d7ca7f064d2bdf606519f9fe1229c381", GitTreeState:"clean", BuildDate:"2017-08-17T08:48:23Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.4", GitCommit:"793658f2d7ca7f064d2bdf606519f9fe1229c381", GitTreeState:"clean", BuildDate:"2017-08-17T08:30:51Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
```

## Logs:
```
kubectl logs -f weave-net-4td10 -c weave-npc
INFO: 2017/08/23 20:36:08.396989 Starting Weaveworks NPC 2.0.3; node name "wshu-docker"
INFO: 2017/08/23 20:36:08.397152 Serving /metrics on :6781
Wed Aug 23 20:36:08 2017 <5> ulogd.c:843 building new pluginstance stack: 'log1:NFLOG,base1:BASE,pcap1:PCAP'
DEBU: 2017/08/23 20:36:08.415114 Got list of ipsets: []
FATA: 2017/08/23 20:36:08.426643 ipset [create weave-local-pods hash:ip comment] failed: ipset v6.29: Unknown argument: `comment'
Try `ipset help' for more information.
: exit status 2
```

## Network:
N/A

### PR DESCRIPTION
**Pull request by mikebryant**
During initialisation try and create an ipset with the comment extension
If this fails, disable comment support

Fixes #3098

### CODE REVIEW

**Comment by mikebryant** [comment]
This hasn't been tested, I don't have something with an old kernel/ipset version

File: npc/ipset/ipset.go [hunk 1]
```diff
@@ -33,22 +33,45 @@ type Interface interface {
...
 func New(logger *log.Logger) Interface {
-	return &ipset{refCount: newRefCount(), Logger: logger}
+	ips := &ipset{refCount: newRefCount(), Logger: logger, enableComments: true}
+
+	// Check for comment support
+	testIpsetName := Name("weave-test-comment")
+	// Clear it out if it already exists
+	_ = ips.Destroy(testIpsetName)
+	// Test for comment support
+	if err := ips.Create(testIpsetName, HashIP); err != nil {
```
Comment (bboreham) [inline npc/ipset/ipset.go]: need to have `enableComments = true` here or the test is invalid

Comment (bboreham) [inline npc/ipset/ipset.go]: oh, sorry, you did do that; ignore me

**Comment by bboreham** [comment]
I tested it on a Centos 7 instance where 2.0.3 is observed to fail.

```
INFO: 2017/08/24 16:27:40.929138 Serving /metrics on :6781
INFO: 2017/08/24 16:27:40.939197 failed to create weave-test-comment; disabling comment support
Thu Aug 24 16:27:40 2017 <5> ulogd.c:843 building new pluginstance stack: 'log1:NFLOG,base1:BASE,pcap1:PCAP'
DEBU: 2017/08/24 16:27:40.958072 Got list of ipsets: []
DEBU: 2017/08/24 16:27:41.077355 EVENT AddPod {"metadata":{"name":"kube-proxy-2z85k","generateName":"kube-proxy-","
namespace":"kube-system","selfLink":"/api/v1/namespaces/kube-system/pods/kube-proxy-[...]c30c942cb6a305d6f2c7c28dd5973"}]}}
INFO: 2017/08/24 16:27:41.109140 creating ipset: &npc.selectorSpec{key:"", selector:labels.internalSelector{}, ipse
tType:"hash:ip", ipsetName:"weave-iuZcey(5DeXbzgRFs8Szo]+@p", nsName:"kube-system"}
```

**Pull request merged**